### PR TITLE
Review fixes

### PR DESCRIPTION
**Pull request by akileng56**
- Remove style max-width: 300
- Add Text style => Header 1, 2,3,4,5,6
- Change Label Color => Positive style, Negative style
- Swap order of Microflow / Page (as in order of onclick prop).
- Fix missing source map

### CODE REVIEW

File: src/ProgressCircle.webmodeler.ts [hunk 1]
```diff
@@ -11,11 +12,22 @@ export class preview extends Component<ContainerProps, {}> {
     }
 
     render() {
-        return createElement(ProgressCircle, this.transformProps(this.props));
+        const { onClickEvent, page, microflow } = this.props;
+        const warnings = ProgressCircleContainer.validateProps(onClickEvent, page, microflow);
```
Comment (Andries-Smit) [inline src/ProgressCircle.webmodeler.ts]: Don't sprint your props. Just pass the `validateProps(this.props)`

File: src/ProgressCircle.xml [hunk 2]
```diff
@@ -5,13 +5,16 @@
...
         <property key="textSize" type="enumeration" defaultValue="medium">
-            <caption>Text size</caption>
+            <caption>Text style</caption>
             <category>Display</category>
             <description>Size of the progress text</description>
             <enumerationValues>
-                <enumerationValue key="small">Small</enumerationValue>
-                <enumerationValue key="medium">Medium</enumerationValue>
-                <enumerationValue key="large">Large</enumerationValue>
+                <enumerationValue key="large">Heading 1</enumerationValue>
```
Comment (Andries-Smit) [inline src/ProgressCircle.xml]: Keeping the key is better for backward compatibility, though it makes you code look ugly.
I don mind you break this part of the client. 

Missing option is `text`

File: src/components/ProgressCircleContainer.ts [hunk 3]
```diff
@@ -77,17 +79,34 @@ export default class ProgressCircleContainer extends Component<ContainerProps, C
         this.subscriptionHandles.forEach((handle) => window.mx.data.unsubscribe(handle));
     }
 
-    private validateProps(): string {
+    public static validateProps(onClickEvent?: OnClickOptions, page?: string, microflow?: string): string {
```
Comment (Andries-Smit) [inline src/components/ProgressCircleContainer.ts]: Don't sprint your props. Just pass the `validateProps(this.props)`

File: src/components/ProgressCircleContainer.ts [hunk 4]
```diff
@@ -77,17 +79,34 @@ export default class ProgressCircleContainer extends Component<ContainerProps, C
...
+                return styleObject;
+            }, {});
+        } catch (error) {
+            console.log("Failed to parse style", style, error);
+        }
+
+        return {};
+    }
+
     private getValue( attribute: string, mxObject?: mendix.lib.MxObject): number | undefined {
```
Comment (Andries-Smit) [inline src/components/ProgressCircleContainer.ts]: Swap signature; getValue(mxObject, attribute)

File: src/components/ProgressCircleContainer.ts [hunk 5]
```diff
@@ -77,17 +79,34 @@ export default class ProgressCircleContainer extends Component<ContainerProps, C
...
+            return style.split(";").reduce<{ [key: string]: string }>((styleObject, line) => {
+                const pair = line.split(":");
+                if (pair.length === 2) {
+                    const name = pair[0].trim().replace(/(-.)/g, match => match[1].toUpperCase());
+                    styleObject[name] = pair[1].trim();
+                }
+                return styleObject;
+            }, {});
+        } catch (error) {
+            console.log("Failed to parse style", style, error);
```
Comment (Andries-Smit) [inline src/components/ProgressCircleContainer.ts]: Add `// tslint:disable-next-line no-console` before

**Review by Andries-Smit — CHANGES_REQUESTED** [review]

File: src/components/ProgressCircleContainer.ts (hunk 4 above)
Comment (akileng56) [reply src/components/ProgressCircleContainer.ts]: TS rule.. an optional parameter can not follow a required one..

File: src/ProgressCircle.webmodeler.ts (hunk 1 above)
Comment (akileng56) [reply src/ProgressCircle.webmodeler.ts]: Had a comment from Johan one time...why pass all the props when you just need three??

Comment (akileng56) [reply src/ProgressCircle.webmodeler.ts]: it goes against information hiding

File: src/components/ProgressCircleContainer.ts (hunk 5 above)
Comment (akileng56) [reply src/components/ProgressCircleContainer.ts]: Done

File: src/components/ProgressCircleContainer.ts (hunk 3 above)
Comment (akileng56) [reply src/components/ProgressCircleContainer.ts]: Done

File: src/ProgressCircle.xml (hunk 2 above)
Comment (akileng56) [reply src/ProgressCircle.xml]: Added

**Pull request merged**